integrations/spark/spark-job-dataquality/spark-job-dataquality-app/src/main/java/com/thinkbiganalytics/spark/dataquality/checker/DataQualityChecker.java

```diff
@@ -20,96 +20,398 @@
+
+        } catch (MissingRuleException e) {
+            log.error("One or more rules are not available: {}", e.getMessage());
+            System.exit(1);
         } catch (Exception e) {
             log.error("Failed to perform data quality checks: {}", e.getMessage());
             System.exit(1);
         }
+    }
 
-        log.info("DataQualityChecker has finished.");
+    public DataQualityChecker() {
+        flowAttributes = new FlowAttributes();
+        ruleList = new ArrayList<DataQualityRule>();
+        availableRules = new HashMap<>();
     }
 
-    public void setArguments(String categoryName, String feedName) {
-        this.categoryName = categoryName;
-        this.feedName = feedName;
+    /**
+     * Add all the available rules Adds all the data quality rules used by the checker
+     */
+    protected void setAvailableRules() {
+        addAvailableRule(new SourceToFeedCountRuleImpl());
+        addAvailableRule(new RowCountRuleImpl());
+        addAvailableRule(new InvalidRowTotalCountRuleImpl());
+        addAvailableRule(new InvalidRowPercentRuleImpl());
     }
 
-    protected HiveContext getHiveContext() {
-        return hiveContext;
+    /**
+     * Sets the active rules based on the available rule list. The flow attribute dq.active.rules is
+     * used to specify active rules. If this attribute does not exist, all available rules are
+     * used.<br>
+     * If any rule does not exist, an exception is thrown
+     * 
+     * @throws MissingRuleException
+     */
+    protected void setActiveRules() throws MissingRuleException {
+
+        String activeRuleStr = flowAttributes.getAttributeValue(DataQualityConstants.DQ_ACTIVE_RULES_ATTRIBUTE,
+                                                                DataQualityConstants.DEFAULT_DQ_ACTIVE_RULES_VALUE);
+
+        if (activeRuleStr.equals(DataQualityConstants.DEFAULT_DQ_ACTIVE_RULES_VALUE)) {
+            log.info("All available Data Quality Rules will be used");
+            ruleList = new ArrayList<>(availableRules.values());
+        } else {
+            String[] parsedRuleStr = activeRuleStr.split(",");
+
+            for (String str : parsedRuleStr) {
+                addDataQualityRule(getAvailableRule(str.trim()));
+            }
+        }
+    }
+
+    /**
+     * Main method that conducts the data quality check. This is done by getting row counts and
+     * iterate through each data quality check. The results summary is provided after all are
+     * executed. The results are written to log as well as hive
+     */
+    protected boolean doDataQualityChecks() {
+
+        boolean isSuccessful = false;
+        try {
+            String databaseName = flowAttributes.getAttributeValue(DataQualityConstants.CATEGORY_ATTRIBUTE);
+            String tableName = flowAttributes.getAttributeValue(DataQualityConstants.FEED_ATTRIBUTE);
+            String processing_dttm = flowAttributes.getAttributeValue(DataQualityConstants.PROCESSING_DTTM_ATTRIBUTE);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00117)